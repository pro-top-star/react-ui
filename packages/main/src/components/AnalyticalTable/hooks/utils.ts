export const makeTemplateColumns = (columns, resizedColumns) => {
  return columns
    .map((column) => {
      if (resizedColumns[column.id]) {
        return `${resizedColumns[column.id]}px`;
      }
      return `minmax(${column.minWidth}px, ${column.width || '1fr'})`;
    })
    .join(' ');
};
