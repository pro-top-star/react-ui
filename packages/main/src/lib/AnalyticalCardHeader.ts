import { AnalyticalCardHeader } from '../components/AnalyticalCard/header/AnalyticalCardHeader';

export { AnalyticalCardHeader };
