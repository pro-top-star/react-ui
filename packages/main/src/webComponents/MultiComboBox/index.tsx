import { Event } from '@ui5/webcomponents-react-base/lib/Event';
import { ValueState } from '@ui5/webcomponents-react/lib/ValueState';
import { withWebComponent } from '@ui5/webcomponents-react/lib/withWebComponent';
import '@ui5/webcomponents/dist/features/InputSuggestions';
import UI5MultiComboBox from '@ui5/webcomponents/dist/MultiComboBox';
import React, { FC, ReactNode } from 'react';
import { WithWebComponentPropTypes } from '../../internal/withWebComponent';

export interface MultiComboBoxPropTypes extends WithWebComponentPropTypes {
  value?: string; // @generated
  placeholder?: string; // @generated
  allowCustomValues?: boolean; // @generated
  disabled?: boolean; // @generated
  valueState?: ValueState; // @generated
  readonly?: boolean; // @generated
  onChange?: (event: Event) => void; // @generated
  onInput?: (event: Event) => void; // @generated
  onSelectionChange?: (event: Event) => void; // @generated
  children?: ReactNode[]; // @generated
}

/**
 * <code>import { MultiComboBox } from '@ui5/webcomponents-react/lib/MultiComboBox';</code>
 * <br />
 * <a href="https://sap.github.io/ui5-webcomponents/playground/components/MultiComboBox" target="_blank">UI5 Web Components Playground</a>
 */
const MultiComboBox: FC<MultiComboBoxPropTypes> = withWebComponent<MultiComboBoxPropTypes>(UI5MultiComboBox);

MultiComboBox.displayName = 'MultiComboBox';

MultiComboBox.defaultProps = {
  value: '', // @generated
  placeholder: '', // @generated
  valueState: ValueState.None // @generated
};

export { MultiComboBox };
