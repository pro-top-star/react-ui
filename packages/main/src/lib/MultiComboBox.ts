import { MultiComboBox } from '../webComponents/MultiComboBox';

export { MultiComboBox };
