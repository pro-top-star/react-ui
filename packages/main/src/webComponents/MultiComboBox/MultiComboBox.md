[Link to UI5 Web Components Playground](https://sap.github.io/ui5-webcomponents/playground/components/MultiComboBox)

```jsx
import { MultiComboBox } from '@ui5/webcomponents-react/lib/MultiComboBox';
```
