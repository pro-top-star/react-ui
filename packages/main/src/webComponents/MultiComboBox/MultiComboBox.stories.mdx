import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs/blocks';
import { MultiComboBox } from '@ui5/webcomponents-react/lib/MultiComboBox';
import { ValueState } from '@ui5/webcomponents-react/lib/ValueState';
import '@ui5/webcomponents-icons/dist/icons/employee.js';
import { MultiComboBoxItem } from '@ui5/webcomponents-react/lib/MultiComboBoxItem';
import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';
import { DocsHeader } from '@shared/stories/DocsHeader';
import { DocsCommonProps } from '@shared/stories/DocsCommonProps';

<Meta
  title="UI5 Web Components / MultiComboBox"
  component={MultiComboBox}
  subcomponents={{ MultiComboBoxItem }}
  argTypes={{
    ...createSelectArgTypes({ valueState: ValueState }),
    ...DocsCommonProps,
    children: { control: { disable: true } },
    icon: { control: { disable: true } },
    valueStateMessage: { control: { disable: true } }
  }}
  args={{
    valueState: ValueState.None,
    style: {},
    className: '',
    tooltip: '',
    slot: '',
    ref: null
  }}
/>

<DocsHeader />

<br />

## Example

<Canvas>
  <Story name="Default">
    {(args) => {
      return (
        <MultiComboBox {...args}>
          <MultiComboBoxItem text="Item 1" />
          <MultiComboBoxItem text="Item 2" />
          <MultiComboBoxItem text="Item 3" />
          <MultiComboBoxItem text="Item 4" />
          <MultiComboBoxItem text="Item 5" />
        </MultiComboBox>
      );
    }}
  </Story>
</Canvas>

## Properties

<ArgsTable story="." />
<div style={{ fontFamily: 'var(--sapFontFamily)', fontSize: 'var(--sapFontSize)', color: 'var(--sapTextColor)' }}>
  <h3>Structure</h3>
  The <code>MultiComboBox</code> consists of the following elements:
  <ul>
    <li>Tokenizer - a list of tokens with selected options.</li>
    <li>Input field - displays the selected option/s as token/s. Users can type to filter the list.</li>
    <li>Drop-down arrow - expands\collapses the option list.</li>
    <li>Option list - the list of available options.</li>
  </ul>
</div>
