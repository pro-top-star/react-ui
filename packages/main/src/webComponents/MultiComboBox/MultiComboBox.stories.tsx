import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';
import { MultiComboBox } from '@ui5/webcomponents-react/lib/MultiComboBox';
import { MultiComboBoxItem } from '@ui5/webcomponents-react/lib/MultiComboBoxItem';
import { ValueState } from '@ui5/webcomponents-react/lib/ValueState';
import React from 'react';

export default {
  title: 'UI5 Web Components / MultiComboBox',
  component: MultiComboBox,
  parameters: {
    subcomponents: { MultiComboBoxItem }
  },
  argTypes: {
    ...createSelectArgTypes({ valueState: ValueState }),
    children: {
      type: null
    },
    ref: {
      type: null
    }
  },
  arg: {
    valueState: ValueState.None
  }
};

export const generatedDefaultStory = (props) => (
  <MultiComboBox
    allowCustomValues={props.allowCustomValues}
    disabled={props.disabled}
    open={props.open}
    placeholder={props.placeholder}
    readonly={props.readonly}
    required={props.required}
    value={props.value}
    valueState={props.valueState}
    onChange={props.onChange}
    onInput={props.onInput}
    onOpenChange={props.onOpenChange}
    onSelectionChange={props.onSelectionChange}
  >
    <MultiComboBoxItem text="Item 1" />
    <MultiComboBoxItem text="Item 2" />
    <MultiComboBoxItem text="Item 3" />
    <MultiComboBoxItem text="Item 4" />
    <MultiComboBoxItem text="Item 5" />
  </MultiComboBox>
);

generatedDefaultStory.storyName = 'Generated default story';
