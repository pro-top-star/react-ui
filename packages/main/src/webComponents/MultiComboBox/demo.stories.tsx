import { action } from '@storybook/addon-actions';
import { boolean, select, text } from '@storybook/addon-knobs';
import { MultiComboBox } from '@ui5/webcomponents-react/lib/MultiComboBox';
import { StandardListItem } from '@ui5/webcomponents-react/lib/StandardListItem';
import { ValueState } from '@ui5/webcomponents-react/lib/ValueState';
import React from 'react';

export default {
  title: 'UI5 Web Components / MultiComboBox',
  component: MultiComboBox,
  parameters: {
    subcomponents: { StandardListItem }
  }
};

export const defaultStory = () => (
  <MultiComboBox
    placeholder={text('placeholder', 'Placeholder...')}
    allowCustomValues={boolean('allowCustomValues', false)}
    disabled={boolean('disabled', false)}
    valueState={select('valueState', ValueState, ValueState.None)}
    readonly={boolean('readonly', false)}
    onChange={action('onChange')}
    onInput={action('onInput')}
    onSelectionChange={action('onSelectionChange')}
  >
    <StandardListItem selected>Selection 1</StandardListItem>
    <StandardListItem>Selection 2</StandardListItem>
    <StandardListItem>Selection 3</StandardListItem>
    <StandardListItem>Selection 4</StandardListItem>
    <StandardListItem>Selection 5</StandardListItem>
  </MultiComboBox>
);

defaultStory.story = {
  name: 'Default story'
};
