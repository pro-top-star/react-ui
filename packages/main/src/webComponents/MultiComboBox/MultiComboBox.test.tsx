import { mountThemedComponent } from '@shared/tests/utils';
import { MultiComboBox } from '@ui5/webcomponents-react/lib/MultiComboBox';
import React from 'react';

describe('MultiComboBox', () => {
  test('Basic Test (generated)', () => {
    const wrapper = mountThemedComponent(<MultiComboBox />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
