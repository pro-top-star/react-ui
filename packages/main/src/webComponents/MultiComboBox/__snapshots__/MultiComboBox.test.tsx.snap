// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiComboBox Basic Test (generated) 1`] = `
<ui5-multi-combobox
  placeholder=""
  value=""
  value-state="None"
/>
`;
